proposals/presentation-message-format.md

```diff
@@ -0,0 +1,1330 @@
+
+A Presentation Request is here defined as a document produced by a Verifier which lists the data criteria for a verification event. Similarly, Presentation Response is the proposed format with which to respond to a Presentation Request. By having Presentation Request as part of the VC specification, [software agents](#glossary) can have a common understanding of the data being requested.
+
+It is expected that by using a Presentation Request, a Verifier can request multiple pieces of data from a Holder such that each:
+- Is held by the Holder in a Verifiable Credential or self attested
+- Is described in a known schema or other linked data
+- Facilitates selective disclosure
+    - Can be a single attribute from the known schema
+    - Can be all data from a schema
+- Is from one of a supplied list of valid Issuers
+
+The concept of Presentation Request already defined as part of [Indy/Sovrin](https://github.com/sovrin-foundation/protocol/blob/master/themis/proof-request.md) as follows:
+
+```The message sent by the relying party to the holder describing the verifiable attributes and appropriate conditions (predicates, issuer of attributes, schema of the credentials used, etc) that the holder need to satisfy.```
+
+This proposal adjusts the definition slightly to move away from the concept of exclusively sending a message. Instead the Presentation Request can be sent or otherwise resolved and includes an optional Presentation URL to which a presentation response can be sent.
+
+### Selective Disclosure via Individually Signed Attributes
+Selective disclosure of attributes from a VC is desirable and should be supported by implementations of this Presentation Request specification. The W3C VC specification lists CL Signatures as the means to achieve selective disclosure. 
+
+For solutions which do not use Zero Knowledge Proofs, this document proposes that individually signing attributes on a VC is a valid alternative for selective disclosure. This solution has the advantage of simpler crypto which anyone can reason about. In addition the approach of individually signing attributes allows an issuer to fully create a credential without requiring an additional issuer/holder interaction as required when using a blinded linked/master secret.
+
+A separate proposal to update the W3C VC spec to support individually signed attributes on a credential will also be submitted.
+
+### Self Attestation
+
+Some use cases require a user to send some information which they self attest. The [Credential Manifest](https://github.com/decentralized-identity/credential-manifest/blob/master/explainer.md) project in DIF is one such example.
+
+To specify that self attestation is valid, add a keyword to the Issuer array for a given claim:
+* `SELF`
+* `did:self`
```

Comment:
Could we use the holder DID as the issuer for self-attested instead of a new DID method? (Like a self-signed X509 cert)?

Reply:
If you are generating the presentation request on the fly then yes absolutely. I was trying to provide a way to have a more generic request that allowed the self-attested concept to be codified. For example if the holder was initiating the exchange with a trusted verifier who had produced a static presentation request such as in the Credential Manifest use case.

I agree about not having a new DID method.